src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -462,7 +462,7 @@ class MySimulation : public SimulationHelper<simDim>
         auto fieldE = dc.get< FieldE >( FieldE::getName(), true );
         auto fieldB = dc.get< FieldB >( FieldB::getName(), true );
 
-        if( step != 0 )
+        if( this->restartRequested )
         {
             namespace nvfct = PMacc::nvidia::functors;
 
```

Comment:
The logic before was wrong. 
`step + 1u` is wrong and is equal to the old implementation.
The current version is tested and correct.